lib/matplotlib/pyplot.py

```diff
@@ -1224,15 +1241,24 @@ def subplot(*args, **kwargs):
 
     # First, search for an existing subplot with a matching spec.
     key = SubplotSpec._from_subplot_args(fig, args)
-    ax = next(
-        (ax for ax in fig.axes
-         if hasattr(ax, 'get_subplotspec') and ax.get_subplotspec() == key),
-        None)
 
-    # If no existing axes match, then create a new one.
-    if ax is None:
+    for ax in fig.axes:
+        # if we found an axes at the position sort out if we can re-use it
+        if hasattr(ax, 'get_subplotspec') and ax.get_subplotspec() == key:
+            # if the user passed no kwargs, re-use
+            if kwargs == {}:
```

Comment:
```suggestion
            if not kwargs:
```

but I know it's just a matter of style.

Reply:
I can see arguments either way here.  I want to say "If kwargs is equal to the empty dictionary" which is equivalent to "kwargs is falsy", but I think the `==` version communicates to the reader better the condition they should be thinking when the read it.